index.js

```diff
@@ -1,37 +1,74 @@
 var uuid = require('node-uuid');
 var CRLF = '\r\n';
 
+function compact(array) {
+  var results = [];
+
+  function truthy(element) {
+    return typeof element !== 'undefined' && element !== null;
+  }
+
+  array.forEach(function(element) {
+    if (truthy(element)) {
+      results.push(element);
+    }
+  });
+  return results;
+}
+
 function generate(tuples, options) {
+  if (tuples.length === 0) {
+    // according to rfc1341 there should be at least one encapsulation
+    return undefined;
+  }
+
   options = options || {};
+  var from = options.from;
+  var preamble = options.preamble;
+  var epilogue = options.epilogue;
   var boundary = options.boundary || uuid();
+
+  if (boundary.length > 70) {
+    // rfc defines boundary to be between 1 and 70 characters long.
+    return undefined;
+  }
+
   var headers = [
-    'From: ' + (options.from || ('nobody ' + Date())),
+    'From: ' + (from || ('nobody ' + Date())),
     'MIME-Version: 1.0',
-    'Content-Type:multipart/mixed; boundary=' + boundary
+    'Content-Type: multipart/mixed; boundary="' + boundary + '"',
   ];
 
-  var delimiter = '--' + boundary;
+  var delimiter = CRLF + '--' + boundary;
   var closeDelimiter = delimiter + '--';
 
-  var parts = tuples.map(function(tuple, i) {
+  var encapsulations = tuples.map(function(tuple, i) {
     var mimetype = tuple.mime || 'text/plain';
-    var encoding = tuple.encoding || 'utf8';
+    var encoding = tuple.encoding || 'utf-8';
     var filename = tuple.filename || ('file' + i);
-    var fileHeaders = [
+
+    var headers = [
       'MIME-Version: 1.0',
       'Content-Type: ' + mimetype + '; charset="' + encoding + '"',
       'Content-Transfer-Encoding: 7bit',
       'Content-Disposition: attachment; filename="' + filename + '"',
-    ].join(CRLF) + CRLF;
+    ];
 
-    return [delimiter, fileHeaders, tuple.content].join(CRLF);
+    var bodyPart = headers.join(CRLF) + CRLF + CRLF + tuple.content;
+
+    var encapsulation = delimiter + CRLF + bodyPart;
+
+    return encapsulation;
```

Comment:
Let's just return `delimiter + CRLF + bodyPart`. No need for `encapsulation variable.